lava/magma/core/model/py/ports.py

```diff
@@ -51,27 +57,74 @@ def peek(self) -> np.ndarray:
 
 
 class PyInPortVectorSparse(PyInPort):
+    """Python implementation of Vector Sparse InPort
+    """
+    # sparse port recieves twice, for val and index.
+    # if not sparse port we need to convert to dense,
+    # vice-versa, possible scipy or numpy tools for conversion
     def recv(self) -> ty.Tuple[np.ndarray, np.ndarray]:
-        pass
+        # Draf Impl. Receives Vector from Sparse
+        #
+        # if not from PyOutPortVectorSparse we need to
+        # process the data received
+        if self._csp_port:
+            return (self._csp_port.recv(),
+                    self._csp_port.recv())
+        else:
+            return (np.zeros(self._shape, self._d_type),
+                    np.zeros(self._shape, self._d_type))
 
     def peek(self) -> ty.Tuple[np.ndarray, np.ndarray]:
-        pass
+        if self._csp_port:
+            return (self._csp_port.peek(),
```

Comment:
This is incorrect. It might not be possible to do a peek twice. Let's talk to Harry and Andreas on how to solve this.

Reply:
Peeks don't remove the data. Hence, you can call it as often as you want.